packages/enzyme/src/ShallowWrapper.js

```diff
@@ -1531,6 +1619,60 @@ class ShallowWrapper {
   }
 }
 
+/**
+ * Updates the context of the primary wrapper when the
+ * `wrappingComponent` re-renders.
+ */
+function updatePrimaryRootContext(wrappingComponent) {
+  const context = getContextFromWrappingComponent(
+    wrappingComponent,
+    getAdapter(wrappingComponent[OPTIONS]),
+  );
+  wrappingComponent[PRIMARY_WRAPPER].setContext({
+    ...wrappingComponent[PRIMARY_WRAPPER][OPTIONS].context,
+    ...context,
+  });
+}
+
+/**
+ * A *special* "root" wrapper that represents the component passed as `wrappingComponent`.
+ * It is linked to the primary root such that updates to it will update the primary.
+ *
+ * @class WrappingComponentWrapper
+ */
+class WrappingComponentWrapper extends ShallowWrapper {
```

Comment:
since this is still accessible via `ShallowWrapper.prototype.getWrappingComponent`, let's move it to its own file?

Reply:
I don't think that's feasible because it would create an import cycle. `./ShallowWrapper` would try to import `./WrappingComponentWrapper` so that it can `new WrappingComponentWrapper()` in the `ShallowWrapper`'s `constructor`. And `./WrappingComponentWrapper` would try to import `./ShallowWrapper` so it can `extends ShallowWrapper`.

And we know that `ShallowWrapper` needs to be defined before we can do `extends ShallowWrapper` (but we can safely `new WrappingComponentWrapper()` in `ShallowWrapper`'s `constructor` even if it's defined down here.) At least if they're in the same file we can ensure that's the case.

Reply:
ah, that's a good point.